### PR TITLE
feat(cli): improve error message with invalid target for `build`

### PR DESCRIPTION
**Pull request by zacharyftw**
### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [x] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Comment by zacharyftw** [comment]
This updated part of code splits the target triple into components using split('-') and puts them into a vector. Then, it checks if the vector has the expected number of components (in this case three). If not, it prints an error message to the console and returns an empty Hashmap.

File: tooling/cli/src/interface/rust.rs [hunk 1]
```diff
@@ -228,8 +228,16 @@ impl Interface for Rust {
...
 
-    let mut s = self.app_settings.target_triple.split('-');
-    let (arch, _, host) = (s.next().unwrap(), s.next().unwrap(), s.next().unwrap());
+    let target_triple = self.app_settings.target_triple.clone();
+    let target_components: Vec<&str> = target_triple.split('-').collect();
+
+    let (arch, _, host) = match target_components.as_slice() {
+      [arch, _, host] => (arch, _, host),
+      _ => {
+        eprintln!("Invalid Target Tuple: {}", target_triple);
```
Comment (amrbashir) [inline tooling/cli/src/interface/rust.rs]:
```suggestion
        log::warn!("Invalid target triple: {}", target_triple);
```

File: tooling/cli/src/interface/rust.rs [hunk 2]
```diff
@@ -228,8 +228,16 @@ impl Interface for Rust {
       self.app_settings.target_triple.clone(),
     );
 
-    let mut s = self.app_settings.target_triple.split('-');
-    let (arch, _, host) = (s.next().unwrap(), s.next().unwrap(), s.next().unwrap());
+    let target_triple = self.app_settings.target_triple.clone();
```
Comment (amrbashir) [inline tooling/cli/src/interface/rust.rs]: I think you shouldn't need to clone here
```suggestion
    let target_triple = self.app_settings.target_triple;
```

**Review by amrbashir — CHANGES_REQUESTED** [review]

**Review by amrbashir — DISMISSED** [review]

**Review by amrbashir — APPROVED** [review]

**Comment by amrbashir** [comment]
Thank you

**Pull request merged**